src/pymodaq/dashboard.py

```diff
@@ -1108,11 +1097,44 @@ def set_preset_mode(self, filename):
                 self.settings_menu.setEnabled(True)
                 self.update_init_tree()
 
-                self.preset_loaded_signal.emit(True)
+            # If the preset is configured to use the pid extension
+            if self.preset_manager.preset_params.child('use_pid').value():
+                self.load_pid_module()
+
+                self.pid_module.settings.child('models', 'model_class').setValue(
+                    self.preset_manager.preset_params.child('pid_models').value())
+                QtWidgets.QApplication.processEvents()
+                self.pid_module.ini_model()
+
+                QtWidgets.QApplication.processEvents()
+
+                for child in putils.iter_children_params(self.preset_manager.preset_params.child('model_settings'),
+                                                         []):
+                    preset_path = self.preset_manager.preset_params.child('model_settings').childPath(child)
+                    path = ['models', 'model_params']
+                    path.extend(preset_path)
+                    self.pid_module.settings.child(*path).setValue(child.value())
+
+                model_class = utils.get_models(
+                    self.preset_manager.preset_params.child('pid_models').value())['class']
+
+                # Add a mock actuator module that will control each setpoint defined in the model.
+                for setp in model_class.setpoints_names:
+                    self.add_move(setp, None, 'PID', [], [], actuators_modules)
+                    actuators_modules[-1].controller = dict(
+                        curr_point=self.pid_module.curr_points_signal,
+                        setpoint=self.pid_module.setpoints_signal,
+                        emit_curr_points=self.pid_module.emit_curr_points_sig)
+                    actuators_modules[-1].ui.IniStage_pb.click()
+                    QtWidgets.QApplication.processEvents()
+                    self.poll_init(actuators_modules[-1])
+                    QtWidgets.QApplication.processEvents()
```

Comment:
ok you've put it there, could you tell me why?

Reply:
Yes it is because in the case you declare to use the PID extension in your preset, the actuators and detectors should be loaded before loading the pid extension.

Reply:
It corresponds to this commit 39d826e88588cd752700edfa96b47799377a9bdd.